### PR TITLE
Handle cache_timeout = 0 in GCSFileSystem

### PR DESCRIPTION
**Pull request by amyewang**
Fixes https://github.com/fsspec/gcsfs/issues/645

### CODE REVIEW

**Comment by martindurant** [comment]
A test for this would be nice.

**Comment by amyewang** [comment]
@martindurant added a test to `test_expiry_keyword` & passes as expected. let me know if you think it should live somewhere else!

File: gcsfs/tests/test_core.py [hunk 1]
```diff
@@ -1428,6 +1428,8 @@ def test_expiry_keyword():
     assert gcs.dircache.listings_expiry_time == 1
     gcs = GCSFileSystem(cache_timeout=1, token="anon")
     assert gcs.dircache.listings_expiry_time == 1
+    gcs = GCSFileSystem(cache_timeout=0, token="anon")
+    assert gcs.dircache.listings_expiry_time == 0
```
Comment (martindurant) [inline gcsfs/tests/test_core.py:1432]: Does this imply that the cache is disabled? What was the previous value before this PR, None?

Comment (amyewang) [reply gcsfs/tests/test_core.py:1432]: Yes, based on the docstring the `listings_expiry_time` should be set to 0, which means cache has a timeout of 0. Before this change the `gcs.dircache.listings_expiry_time` evaluated to `None`, so the cache was enabled

**Pull request merged**